### PR TITLE
fix child component data access for highlighting

### PR DESCRIPTION
**Pull request by rgottsc**
- sending searchTerm Data via @input decorator

- add highlight to child component scss

- fix highlight.pipe.ts: fix regex

- refactor mapping-data-sources.pipe.ts

### CODE REVIEW

File: projects/wave-core/src/lib/util/pipes/highlight.pipe.ts [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
 
-    // TODO: replace <span> with a <higlight> component?
-    transform(text: string, term: string): string {
+export class HighlightPipe implements PipeTransform {
+    transform(text: string, search): string {
+        if (search && text) {
+            let pattern = search.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&');
+            pattern = pattern.split(' ').filter((t) => {
+                return t.length > 0;
+            }).join('|');
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/util/pipes/highlight.pipe.ts:11]: What do you accomplish in comparison to the previous functionality by replacing something on the search term and by splitting it into several parts?

File: projects/wave-core/src/lib/util/pipes/highlight.pipe.ts [hunk 2]
```diff
@@ -1,22 +1,19 @@
-import {Pipe, PipeTransform} from '@angular/core';
+import { PipeTransform, Pipe } from '@angular/core';
 
 @Pipe({name: 'waveHighlightPipe'})
-export class HighlightPipe implements PipeTransform {
 
-    // TODO: replace <span> with a <higlight> component?
-    transform(text: string, term: string): string {
+export class HighlightPipe implements PipeTransform {
+    transform(text: string, search): string {
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/util/pipes/highlight.pipe.ts:6]:
```suggestion
    transform(text: string, searchTerm: string): string {
```
You stripped the data type of the search term, but I guess it is always a string.

File: projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts [hunk 3]
```diff
@@ -28,8 +28,8 @@ import {NotificationService} from '../../../../notification.service';
 export class VectorSourceDatasetComponent implements OnInit {
 
     @Input() dataset: MappingSource;
-
-    searchTerm: string; // TODO: this was needed to get prod build working...
+    @Input() searchTerm: string;
+    // searchTerm: string; // TODO: this was needed to get prod build working...
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts:32]: You can delete the commented out code.

File: projects/wave-core/src/lib/operators/dialogs/data-repository/raster/source-dataset.component.ts [hunk 4]
```diff
@@ -29,8 +29,8 @@ import {Colormap} from '../../../../colors/colormaps/colormap.model';
 export class SourceDatasetComponent implements OnInit {
 
     @Input() dataset: MappingSource;
-
-    searchTerm: string; // TODO: this was needed to get prod build working...
+    @Input() searchTerm: string;
+    // searchTerm: string; // TODO: this was needed to get prod build working...
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/dialogs/data-repository/raster/source-dataset.component.ts:33]: You can delete the commented out code.

File: projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.scss [hunk 5]
```diff
@@ -16,6 +16,10 @@ mat-list-item ::ng-deep .highlight {
     color: var(--wave-accent-color, grey);
     text-decoration: underline;
 }
+.highlight {
+    color: var(--wave-accent-color, grey);
+    text-decoration: underline;
+}
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.scss:22]: This code is subsumed by 
```
.highlight,
mat-list-item ::ng-deep .highlight {
    color: var(--wave-accent-color, grey);
    text-decoration: underline;
}
```
from the four lines above, so you should be able to delete it.

File: projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html [hunk 6]
```diff
@@ -18,13 +18,12 @@
     </mat-menu>
     -->
 </wave-sidenav-search>
-
 <mat-list *ngIf="(sources | async).length > 0; else noSources">
     <ng-template ngFor let-source [ngForOf]="sources | async | waveMappingDataSourceFilter:searchTerm">
-        <wave-vector-source-dataset *ngIf="source.operator === 'ogr_source'; else rasterDataset" [dataset]="source">
+        <wave-vector-source-dataset *ngIf="source.operator === 'ogr_source'; else rasterDataset" [dataset]="source" [searchTerm]=searchTerm>
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html:23]:
```suggestion
        <wave-vector-source-dataset *ngIf="source.operator === 'ogr_source'; else rasterDataset" [dataset]="source" [searchTerm]="searchTerm">
```

Although it seems to work, you should use quotation marks around these properties.

File: projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html [hunk 7]
```diff
@@ -18,13 +18,12 @@
...
         </wave-vector-source-dataset>
         <ng-template #rasterDataset>
-            <wave-source-dataset [dataset]="source"></wave-source-dataset>
+            <wave-source-dataset [dataset]="source" [searchTerm]=searchTerm></wave-source-dataset>
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html:26]:
```suggestion
            <wave-source-dataset [dataset]="source" [searchTerm]="searchTerm"></wave-source-dataset>
```

Although it seems to work, you should use quotation marks around these properties.

File: projects/wave-core/src/lib/operators/dialogs/data-repository/raster/source-dataset.component.html [hunk 8]
```diff
@@ -28,7 +28,8 @@
             <ng-container matColumnDef="name">
                 <mat-header-cell *matHeaderCellDef>Channel</mat-header-cell>
                 <mat-cell *matCellDef="let channel">
-                    {{channel.name}}
+                    <span bind-innerHtml="channel.name | waveHighlightPipe:searchTerm | waveSafeHtml"></span>
+                    <!--{{channel.name}}-->
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/operators/dialogs/data-repository/raster/source-dataset.component.html:32]: Also here, you can delete the commented out code. We can recover it from version control if necessary.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
Looks good so far. There are some minor comments for you to address.

File: projects/wave-core/src/lib/util/pipes/highlight.pipe.ts (hunk 1 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/util/pipes/highlight.pipe.ts]: changed back to old regex

File: projects/wave-core/src/lib/util/pipes/highlight.pipe.ts (hunk 2 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/util/pipes/highlight.pipe.ts]: does not exist anymore in this form

File: projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts (hunk 3 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts]: deleted

File: projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.scss (hunk 5 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.scss]: deleted

File: projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html (hunk 6 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html]: added question marks

File: projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html (hunk 7 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/operators/dialogs/data-repository/data-repository.component.html]: added question marks

File: projects/wave-core/src/lib/operators/dialogs/data-repository/raster/source-dataset.component.html (hunk 8 above)
Comment (rgottsc) [reply projects/wave-core/src/lib/operators/dialogs/data-repository/raster/source-dataset.component.html]: deleted

**Review by ChristianBeilschmidt — APPROVED** [review]

**Comment by ChristianBeilschmidt** [comment]
https://github.com/umr-dbs/wave/issues/81

**Pull request merged**